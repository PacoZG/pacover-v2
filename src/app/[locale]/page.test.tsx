/* eslint-disabled */
import React from 'react'
import { render } from '@testing-library/react'
import { beforeEach, describe, expect, test, vi } from 'vitest'
import Page from '@/app/[locale]/page'

describe('Main Page', () => {
  describe('given configured with dependencies', () => {
    beforeEach(() => {
      window.scrollTo = vi.fn()

      vi.mock('next-intl', () => ({
        useTranslations: () => (key: string) => key,
      }))
    })

    describe('Main Page component', () => {
      test('matches the snapshot', () => {
        const { container } = render(<Page />)

        expect(container).toMatchSnapshot()
      })
    })
  })
})
