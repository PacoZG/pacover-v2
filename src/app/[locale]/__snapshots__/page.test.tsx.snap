// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main Page > given configured with dependencies > Main Page component > matches the snapshot 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-400 dark:bg-gray-900 text-gray-800 dark:text-gray-200 transition-colors duration-500"
  >
    <div
      class="relative h-screen flex items-center justify-center text-center"
    >
      <img
        alt="Professional Coding Environment"
        class="absolute inset-0 w-full h-full object-cover z-0"
        data-nimg="1"
        decoding="async"
        height="96"
        loading="lazy"
        src="https://placehold.co/1920x1080/4F46E5/ffffff?text=Professional+Coding+Environment"
        style="color: transparent;"
        width="96"
      />
      <div
        class="absolute inset-0 bg-black bg-opacity-60 dark:bg-opacity-70 z-10"
      />
      <div
        class="relative z-20 max-w-4xl px-4 py-8 rounded-xl bg-gray-600 dark:bg-gray-900 bg-opacity-70 dark:bg-opacity-70 backdrop-blur-sm shadow-lg"
      >
        <h1
          class="text-4xl sm:text-5xl md:text-6xl font-extrabold text-white leading-tight"
        >
          hero.headline
        </h1>
        <p
          class="mt-4 text-lg sm:text-xl md:text-2xl text-gray-200"
        >
          hero.subheadline
        </p>
        <a
          class="mt-8 inline-block px-8 py-3 text-lg font-semibold text-white bg-blue-600 rounded-lg shadow-xl hover:bg-blue-700 transition duration-300"
          data-testid="hero-link"
          href="#contact-form"
        >
          hero.button
        </a>
      </div>
    </div>
    <div
      class="py-20 flex justify-center"
    >
      <div
        class="w-full sm:w-11/12 lg:w-9/12 px-4"
      >
        <h2
          class="text-3xl sm:text-4xl font-bold text-center mb-12"
        >
          services.headline
        </h2>
        <div
          class="grid grid-cols-1 sm:grid-cols-2 xl:grid-cols-4 gap-8"
        >
          <div
            class="bg-gray-300 dark:bg-gray-800 p-8 rounded-xl shadow-lg hover:shadow-2xl transform hover:-translate-y-2 transition duration-300 text-center"
          >
            <div
              class="flex items-center justify-center p-4 rounded-full bg-blue-600 mb-4 mx-auto w-20 h-20"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-globe h-10 w-10 text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <path
                  d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
                />
                <path
                  d="M2 12h20"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              services.simple_sites.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              services.simple_sites.description
            </p>
          </div>
          <div
            class="bg-gray-300 dark:bg-gray-800 p-8 rounded-xl shadow-lg hover:shadow-2xl transform hover:-translate-y-2 transition duration-300 text-center"
          >
            <div
              class="flex items-center justify-center p-4 rounded-full bg-blue-600 mb-4 mx-auto w-20 h-20"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-server h-10 w-10 text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="8"
                  rx="2"
                  ry="2"
                  width="20"
                  x="2"
                  y="2"
                />
                <rect
                  height="8"
                  rx="2"
                  ry="2"
                  width="20"
                  x="2"
                  y="14"
                />
                <line
                  x1="6"
                  x2="6.01"
                  y1="6"
                  y2="6"
                />
                <line
                  x1="6"
                  x2="6.01"
                  y1="18"
                  y2="18"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              services.full_stack_apps.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              services.full_stack_apps.description
            </p>
          </div>
          <div
            class="bg-gray-300 dark:bg-gray-800 p-8 rounded-xl shadow-lg hover:shadow-2xl transform hover:-translate-y-2 transition duration-300 text-center"
          >
            <div
              class="flex items-center justify-center p-4 rounded-full bg-blue-600 mb-4 mx-auto w-20 h-20"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-cloud h-10 w-10 text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.5 19H9a7 7 0 1 1 6.71-9h1.79a4.5 4.5 0 1 1 0 9Z"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              services.devops.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              services.devops.description
            </p>
          </div>
          <div
            class="bg-gray-300 dark:bg-gray-800 p-8 rounded-xl shadow-lg hover:shadow-2xl transform hover:-translate-y-2 transition duration-300 text-center"
          >
            <div
              class="flex items-center justify-center p-4 rounded-full bg-blue-600 mb-4 mx-auto w-20 h-20"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-settings h-10 w-10 text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.671 4.136a2.34 2.34 0 0 1 4.659 0 2.34 2.34 0 0 0 3.319 1.915 2.34 2.34 0 0 1 2.33 4.033 2.34 2.34 0 0 0 0 3.831 2.34 2.34 0 0 1-2.33 4.033 2.34 2.34 0 0 0-3.319 1.915 2.34 2.34 0 0 1-4.659 0 2.34 2.34 0 0 0-3.32-1.915 2.34 2.34 0 0 1-2.33-4.033 2.34 2.34 0 0 0 0-3.831A2.34 2.34 0 0 1 6.35 6.051a2.34 2.34 0 0 0 3.319-1.915"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              services.maintenance.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              services.maintenance.description
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="py-20 bg-gray-300 dark:bg-gray-800 flex justify-center"
    >
      <div
        class="w-full sm:w-11/12 lg:w-9/12 px-4"
      >
        <h2
          class="text-3xl sm:text-4xl font-bold text-center mb-12"
        >
          process.headline
        </h2>
        <div
          class="grid grid-cols-1 sm:grid-cols-2 xl:grid-cols-4 gap-8"
        >
          <div
            class="p-6 rounded-xl border-2 border-gray-400 dark:border-gray-700 shadow-md text-center"
          >
            <div
              class="flex items-center justify-center mb-4 mx-auto"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-handshake h-10 w-10 text-blue-500 dark:text-blue-300"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m11 17 2 2a1 1 0 1 0 3-3"
                />
                <path
                  d="m14 14 2.5 2.5a1 1 0 1 0 3-3l-3.88-3.88a3 3 0 0 0-4.24 0l-.88.88a1 1 0 1 1-3-3l2.81-2.81a5.79 5.79 0 0 1 7.06-.87l.47.28a2 2 0 0 0 1.42.25L21 4"
                />
                <path
                  d="m21 3 1 11h-2"
                />
                <path
                  d="M3 3 2 14l6.5 6.5a1 1 0 1 0 3-3"
                />
                <path
                  d="M3 4h8"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              process.discovery.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              process.discovery.description
            </p>
          </div>
          <div
            class="p-6 rounded-xl border-2 border-gray-400 dark:border-gray-700 shadow-md text-center"
          >
            <div
              class="flex items-center justify-center mb-4 mx-auto"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-code h-10 w-10 text-blue-500 dark:text-blue-300"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m16 18 6-6-6-6"
                />
                <path
                  d="m8 6-6 6 6 6"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              process.development.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              process.development.description
            </p>
          </div>
          <div
            class="p-6 rounded-xl border-2 border-gray-400 dark:border-gray-700 shadow-md text-center"
          >
            <div
              class="flex items-center justify-center mb-4 mx-auto"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-cloud h-10 w-10 text-blue-500 dark:text-blue-300"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.5 19H9a7 7 0 1 1 6.71-9h1.79a4.5 4.5 0 1 1 0 9Z"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              process.deployment.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              process.deployment.description
            </p>
          </div>
          <div
            class="p-6 rounded-xl border-2 border-gray-400 dark:border-gray-700 shadow-md text-center"
          >
            <div
              class="flex items-center justify-center mb-4 mx-auto"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-circle-check-big h-10 w-10 text-blue-500 dark:text-blue-300"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.801 10A10 10 0 1 1 17 3.335"
                />
                <path
                  d="m9 11 3 3L22 4"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold mb-2"
            >
              process.support.title
            </h3>
            <p
              class="text-gray-700 dark:text-gray-400"
            >
              process.support.description
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="py-20 flex justify-center flex-col items-center relative"
    >
      <div
        class="w-full sm:w-11/12 lg:w-9/12 px-4 text-center"
      >
        <h2
          class="text-3xl sm:text-4xl font-bold leading-tight"
        >
          cta.headline
        </h2>
        <p
          class="mt-4 text-lg sm:text-xl text-gray-700 dark:text-gray-400"
        >
          cta.subheadline
        </p>
        <a
          class="mt-8 inline-block px-10 py-4 text-xl font-semibold text-white bg-blue-600 rounded-lg shadow-xl hover:bg-blue-700 transition duration-300"
          data-testid="action-link"
          href="#contact-form"
        >
          cta.button
        </a>
      </div>
    </div>
  </div>
</div>
`;
